09_maps/main.go

```diff
@@ -111,18 +122,261 @@ func main() {
+	var heightConverted float64 = float64(heightConvert)
+	var weightConvert int = mapInput["weight"]
+	var weightConverted float64 = float64(weightConvert)
+
+	// var result float64 = bmi
+
+	value, exists = mapInput["height"]
+	if !exists || value == 0 {
+		fmt.Println("Height not found")
+	}
+	value, exists = mapInput["weight"]
+	if !exists || value == 0 {
+		fmt.Println("Weight not found")
+	}
+	// debug
+	// fmt.Println("weight debug", mapInput["weight"])
+	// fmt.Println("height debug", mapInput["height"])
+	// debug
+	// fmt.Println(heightAdjust)
+	// fmt.Println(weightConverted)
+	bmi = (weightConverted / (heightConverted * heightConverted))
+	bmi = bmi * 10000
+	// debug
+	// fmt.Println(bmi)
+	// TODO: check and fix syntax for name and bmi variable
+	// fmt.Println("BMI is", bmi)
+	return bmi
+
+	// for {
+	// 	var counter int = 0
+	// 	if counter > len(mapInput) {
+	// 		break
+	// 	}
+	// 	counter++
+	// 	return bmi
+	// }
+
+}
+
+// TODO: Create a function that takes an array of maps as an argument and
+// prints the average BMI of the people in the array.
+
+// create function
+// accept array of maps as input
+// loop through all arrays (e.g. "people" array of maps)
+// calculate BMI (calcBMI function)
+// add the sum of all array elements
+// increase counter
+// divide through the amount of array elements
+// print result = avg BMI
+func calcAvgBMI(arrayOfMapInput []map[string]int) float64 {
+	var counter int
+	var bmiSum float64
+	var avgBMI float64
+	var lenConvert int = len(arrayOfMapInput)
+	var lenConverted float64 = float64(lenConvert)
+	for {
+		if counter >= len(arrayOfMapInput) {
+			break
+		}
+		bmiSum += calcBMI(arrayOfMapInput[counter])
+		// debug
+		// fmt.Println(bmiSum)
+		counter++
+	}
+	// debug
+	// fmt.Println(lenConverted)
+	avgBMI = bmiSum / lenConverted
+	// debug
+	// fmt.Println("avg debug print", avgBMI)
+	return avgBMI
+}
+
+// TODO: Create a function that takes an array of maps as an argument and
+// returns the map with the highest BMI.
+
+// create function
+// accept array of maps as input
+// calculate the BMI with function calcBMI for all elements of the array
+// compare all elements with each other to get highest value/BMI
+// if one element is bigger than ALL of the others, return that element
+// else continue comparing
+
+// helper functions
+// write a function that takes a list of int and return the biggest, or 0.
+func largestInt(s []int) int {
+	if len(s) == 0 {
+		return 0
+	}
+	var largest = s[0]
+
+	for {
+		var counter int
+		if counter > len(s) {
+			break
+		}
+		if s[counter] > largest {
+			largest = s[counter]
+		}
+		counter++
+	}
+	return largest
+}
+
+// helper functions
+// write a function that takes a list of string and return the longest string, or an empty string.
+func largestString(s []string) string {
+	if len(s) == 0 {
+		return ""
+	}
+	var largest = s[0]
+	for {
+		var counter int
+		if counter > len(s) {
+			break
+		}
+		if s[counter] > largest {
```

Comment:
This is incorrect, you need to compare string length

Reply:
Should be fixed.